src/test/scala/chiselTests/SwitchSpec.scala

```diff
@@ -0,0 +1,23 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import chisel3.util._
+import chisel3.testers.BasicTester
+
+class MultiSwitchFireTester extends BasicTester {
+  val wire = WireInit(0.U)
+  switch (0.U) {
+    is (0.U) { wire := 1.U } // Only this one should happen
+    is (0.U) { wire := 2.U }
```

Comment:
Oddly, this test immediately demonstrates that the new switch has different behavior than the old switch. The old one would run overlapping when conditions (and the last one would win). In the case where 'is' are with literals, this shouldn't change behavior (unless sloppy code) but could be problematic if you ever used a non-literal in the is block.

Reply:
Maybe I wasn't clear in the message . This is intentionally changing behavior and I wrote the test this way because it demonstrated the change.

Reply:
Although I mucked up the message below, I'll fix that